lib/chai/utils/addProperty.js

```diff
@@ -40,8 +39,9 @@ module.exports = function (ctx, name, getter) {
 
   Object.defineProperty(ctx, name,
     { get: function addProperty() {
+        var keep_ssfi = flag(this, 'keep_ssfi');
```

Comment:
Could you also explain why we need the `keep_ssfi` flag. As far as I can see it is used to to override ssfi when we chain on properties - is this right?